packages/admin/admin/src/theme/componentsTheme/MuiDataGrid.tsx

```diff
@@ -79,6 +80,7 @@ export const getMuiDataGrid: GetMuiComponentTheme<"MuiDataGrid"> = (component, {
             },
         },
         localeText: {
+            filterOperatorSearch: <FormattedMessage id="comet.dataGrid.filterOperators.search" defaultMessage="search" />,
```

Comment:
@jamesricky I "fixed" i18n by creating a fake localeText in the theme that's picked up by the Data Grid. What do you think about this? Is there a better way?

Reply:
I'm not really sure what you mean with "fake localeText in the theme that's picked up by the Data Grid" 😅 
What exactly is the issue you're attempting to fix?

Reply:
The hack is that getLocaleText expects `string` and that I passed `ReactNode`, which may break. For instance, MUI could decide to pass the operator label as aria label, which would result in "[Object object]".

Reply:
how do we handle this for other translations (the ones from DataGrid)?

Reply:
> Maybe I could move the intl call to the theme, but I'm not sure if this improves it.

Just had a look at this, this would require us to pass intl to createCometTheme, I don't think we want this.

> how do we handle this for other translations (the ones from DataGrid)?

In our custom components (e.g., [the alternative pagination](https://github.com/vivid-planet/comet/blob/next/packages/admin/admin/src/dataGrid/pagination/DataGridPagination.tsx)) we can use react-intl. Otherwise we use the official localizations from MUI: https://v7.mui.com/x/react-data-grid/localization/.

Reply:
We only added support for it, but never implemented it in App.tsx 🤦🏼‍♂️ 

https://github.com/vivid-planet/comet/pull/2139

Reply:
> we could have createCometTheme({}, { muiLocale: deDE, intl }); or similar.

I don't think we currently have any other use-cases but maybe it makes sense, we might have more use-cases in the future, especially if we customize `DataGrid` further 🤔

Reply:
I'd like to do/try this in a follow-up PR since I don't want to bloat this PR any further. I'v reverted the change to the `dataGridManyToManyColumn` here: ab86bd2666f8ae33ba04ee8a922c108235f89045.